core/src/transport/optional.rs

```diff
@@ -28,7 +29,8 @@ use multiaddr::Multiaddr;
 /// enabled (read: contains `Some`), then dialing and listening will be handled by the inner
 /// transport.
 #[derive(Debug, Copy, Clone)]
-pub struct OptionalTransport<T>(Option<T>);
+#[pin_project::pin_project]
```

Comment:
This is necessary because we don't want to require `Unpin` everywhere, correct?

Reply:
Yes